syntax = "proto3";

package proto;

service ClientService {
    rpc CreateClient (CreateClientRequest) returns (Client);
    rpc GetClients (Empty) returns (ClientsResponse);
    rpc UpdateClient (UpdateClientRequest) returns (Client);
    rpc DeleteClient (DeleteClientRequest) returns (Empty);
}

message Client {
    string id = 1;
    string first_name = 2;
    string last_name = 3;
    string address = 4;
    string phone = 5;
}

message ClientsResponse {
    repeated Client clients = 1;
}

message CreateClientRequest {
    string first_name = 1;
    string last_name = 2;
    string address = 3;
    string phone = 4;
}

message UpdateClientRequest {
    string id = 1;
    string first_name = 2;
    string last_name = 3;
    string address = 4;
    string phone = 5;
}

message DeleteClientRequest {
    string id = 1;
}

message Empty {}
